### PR TITLE
Travis CI: Maven

### PR DESCRIPTION
**Pull request by davidgraeff**
Only log warnings/errors

Signed-off-by: David Graeff <david.graeff@web.de>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -51,4 +51,4 @@ matrix:
   allow_failures:
     - jdk: openjdk11
 script:
-  - mvnp clean install -B -DskipChecks=true -DskipTests=true $(extra_maven_opts)
+  - MAVEN_OPTS="-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=WARN -Dorg.slf4j.simpleLogger.defaultLogLevel=WARN" mvnp clean install -B -DskipChecks=true -DskipTests=true $(extra_maven_opts)
```
Comment (J-N-K) [inline .travis.yml]: '-Dorg.slf4j.simpleLogger.defaultLogLevel=WARN' is not making any difference in output in my local build. The other one works as expected.

Comment (davidgraeff) [reply .travis.yml]: I tested it locally. But with "warn" instead of WARN

**Comment by davidgraeff** [comment]
More build system related:

> 244/245| openHAB Add-ons :: Features :: Karaf :: Add-ons 2.5.0-SNAPSHOT

This steps takes forever. Can we agree on not performing any global operations like karaf feature request repo creation (?) on pull requests and instead setup a cron job once a day on travis and show the status via a badge in the readme (like other projects do it)?.

**Comment by davidgraeff** [comment]
@wborn  This is only a testrun. If it works, I like to remove what we do now with hiding the output and let maven print straight to the console.

Ideally we only see download errors / build errors / test errors and warnings/errors from sat.

**Comment by wborn** [comment]
If the logging configurations are added to the plugin configuration in the POM files everyone (including Jenkins) can use them instead of only Travis. 

> Can we agree on not performing any global operations like karaf feature request repo creation 

The idea of the build is to check for errors to prevent them from entering the code base. These errors are not found when SAT and feature validation is disabled. I think it is a bad idea to merge PRs with issues and hope that someone (else) at some point in time fixes these. It's also frustrating to have a local build fail because some other build didn't find these errors because it used other options.

**Comment by J-N-K** [comment]
I agree with @wborn, neither SAT nor feature check should be disabled.

A lot of the time in the feature check (at least in my local build) is used for downloading unchanged artifacts, Maybe we can optimize that?

**Comment by wborn** [comment]
>  I like to remove what we do now with hiding the output and let maven print straight to the console.

I like that I can now easily get an overview of the build progress without having to filter it myself from lots  scrolling lines. As long as it's building I think no one is interested in all those details.

**Comment by davidgraeff** [comment]
> nor feature check should be disabled.

I don't want to disable them. But the feature check should be a syntax check only and not bundling the entire repo, downloading stuff etc. If a new binding in its own separate directory builds find (which is what I want to archive here) then the entire build will succeed, otherwise our buildsystem is fragile.

What we need to do though is having a feature.xml file per binding and merging those (yes, I have seen a maven plugin that does this), so that all changes are in one place. The bom should also be generated automatically. A new binding PR should really only alter the parent pom.xml and its directory.

> I like that I can now easily get an overview of the build progress

Yes I want that as well. Have a look at the recent travis build. Those are all warnings and errors. We should just fix them or make them disappear. One way is to make SAT not output to the console as I said, because SAT is producing 95% of those warnings.

**Comment by davidgraeff** [comment]
To make the compiler silent for a full build I suggest to add a property "nowarn" to pom.xml. A pull request build will compile with all warnings (maybe even promoted to errors) enabled for that extension specific directory and a full build will be build with "nowarn" as it is quite impossible to fix all those warnings for us.

```
      ...
       <plugin>
          <groupId>org.apache.maven.plugins</groupId>
          <artifactId>maven-compiler-plugin</artifactId>
          <configuration>
           ...
            <compilerArgs>
              <arg>${compiler.warn}</arg>
              <arg>${compiler.error}</arg>
            </compilerArgs>
           ...
        </plugin>
     ...
<profiles>
    <profile>
      <id>compiler-warnings-on</id>
      <activation>
        <property>
          <name>!nowarn</name>
        </property>
      </activation>
      <properties>
        <compiler.warn>-err:+nullAnnot(org.eclipse.jdt.annotation.Nullable|org.eclipse.jdt.annotation.NonNull|org.eclipse.jdt.annotation.NonNullByDefault),+inheritNullAnnot,-nullUncheckedConversion</compiler.warn>
        <compiler.error>-warn:+null,+inheritNullAnnot,+nullAnnotConflict,-nullUncheckedConversion,+nullAnnotRedundant,+nullDereference</compiler.error>
      </properties>
    </profile>
    <profile>
      <id>compiler-warnings-off</id>
      <activation>
        <property>
          <name>nowarn</name>
        </property>
      </activation>
      <properties>
        <compiler.warn>-err:-nullAnnot(org.eclipse.jdt.annotation.Nullable|org.eclipse.jdt.annotation.NonNull|org.eclipse.jdt.annotation.NonNullByDefault),-inheritNullAnnot,-nullUncheckedConversion</compiler.warn>
        <compiler.error>-warn:-null,-inheritNullAnnot,-nullAnnotConflict,-nullUncheckedConversion,-nullAnnotRedundant,-nullDereference</compiler.error>
      </properties>
    </profile>
<profiles>
```

**Comment by wborn** [comment]
> Have a look at the recent travis build.

I think it becomes more difficult to follow progress of a build when it shows all the details of downloads etc. again. You'll also run the risk of the Travis build [being terminated again by exceeding the 4MB log limit](https://docs.travis-ci.com/user/common-build-problems/#log-length-exceeded). 

Ideally I'd have just a high level overview by default. Then if I'm interested in more details I'd like to click something to show more details. A progress bar and ETA would also be nice like [maven-progress-bar](https://github.com/philleonard/maven-progress-bar) does. :-)

**Comment by davidgraeff** [comment]
I just had a look at https://github.com/apache/maven/blob/master/maven-embedder/src/main/java/org/apache/maven/cli/event/ExecutionEventLogger.java where I thought the reactor summary is printed. But they are using this logging context for all the other nonsense in between as well, so it's really tough to make maven less verbose.

But this is really a maven problem and that this source file for example hasn't changed since 5 years. Java developers seem to be used to redirecting mavens verbosity to a file and then grepping through it for useful information.

For now I'd say I'm concentrating on the idea to selectively build on travis and get that cache invalidation fix in.

**Pull request closed without merging**